packages/lesswrong/components/vulcan-forms/FormGroup.tsx

```diff
@@ -47,212 +51,126 @@ const FormGroupHeaderComponent = registerComponent('FormGroupHeader', FormGroupH
+  throwError: any;
+  currentValues: any;
+  updateCurrentValues: any;
+  deletedValues: any[];
+  addToDeletedValues: any;
+  clearFieldErrors: any;
+  formType: "new" | "edit";
+  currentUser: UsersCurrent | null;
+  formProps: any;
+  formComponents: ComponentTypes;
+  setFooterContent: (footerContent: React.ReactNode) => void;
+}
 
-interface FormGroupExternalProps extends FormGroupType<CollectionNameString> {
-  errors: any[]
-  throwError: any
-  currentValues: any
-  updateCurrentValues: any
-  deletedValues: any
-  addToDeletedValues: any
-  clearFieldErrors: any
-  formType: "new"|"edit"
-  currentUser: UsersCurrent|null
-  formComponents: ComponentTypes
-  formProps: any
-  disabled: boolean
+interface FormGroupProps extends FormGroupSafeType<CollectionNameString>, Omit<FormControlProps, 'setFooterContent'> {
   fields: FormField<any>[]
 }
-interface FormGroupProps extends FormGroupExternalProps, WithLocationProps {
-}
-interface FormGroupState {
-  collapsed: boolean
-  footerContent: React.ReactNode
-}
-
-class FormGroup extends PureComponent<FormGroupProps,FormGroupState> {
-  constructor(props: FormGroupProps) {
-    super(props);
-    this.toggle = this.toggle.bind(this);
-    this.renderHeading = this.renderHeading.bind(this);
-    this.setFooterContent = this.setFooterContent.bind(this);
-
-    const { query } = this.props.location;
-    const highlightInFields = query.highlightField && props.fields.map(f => f.name).includes(query.highlightField)
-    const collapsed = (props.startCollapsed && !highlightInFields) || false
-
-    this.state = {
-      collapsed,
-      footerContent: null,
-    };
-  }
 
-  toggle() {
-    this.setState({
-      collapsed: !this.state.collapsed
-    });
-  }
-
-  setFooterContent(footerContent: React.ReactNode) {
-    this.setState({ footerContent });
-  }
-
-  renderHeading(FormComponents: ComponentTypes) {
-    const component = <FormComponents.FormGroupHeader
-      toggle={this.toggle}
-      label={this.props.label}
-      collapsed={this.state.collapsed}
-    />
-    if (this.props.helpText) {
-      return <Tooltip title={this.props.helpText}>
-        <span>
-        {component}
-        </span>
-      </Tooltip>
-    }
-    return component
-  }
-
-  // if at least one of the fields in the group has an error, the group as a whole has an error
-  hasErrors = () =>
-    _.some(this.props.fields, (field: FormField<any>) => {
-      return !!this.props.errors.filter((error: any) => error.path === field.path)
-        .length;
-    });
-
-  render() {
-    const { name, fields, formComponents, label, defaultStyle, hideHeader, flexStyle, flexAlignTopStyle, paddingStyle, formProps } = this.props;
-    const { collapsed } = this.state;
-    const FormComponents = mergeWithComponents(formComponents);
-    const groupStyling = !(name === 'default' || defaultStyle)
-    const showHeading = groupStyling && !hideHeader
-
-    return (
-      <FormComponents.FormGroupLayout
+const FormGroup = ({
```

Comment:
The changes in this file are:
- Refactoring `FormGroup` to be a function component
- Supporting LayoutComponents other than the default `FormGroupLayout`
- Some minor changes to what is passed through to `LayoutComponent`/`FormGroupLayout`

Reply:
🙌